packages/alfa-dom/src/get-tab-sequence.ts

```diff
@@ -0,0 +1,46 @@
+import { getTabIndex } from "./get-tab-index";
+import { traverseNode } from "./traverse-node";
+import { Element, NodeType } from "./types";
+
+function binaryIndexSearch(array: Array<Element>, index: number) {
+  let low = 0;
+  let high = array.length;
+
+  if (index === 0) {
+    return high;
+  }
+
+  while (low < high) {
+    const mid = (low + high) >>> 1;
```

Comment:
Another JavaScript gotcha: All bitwise operations operate on 32 bit numbers and will simply cut off bits when numbers exceed 32 bits. The safe, i.e. non-overflowing, way of computing the middle between two array indices is therefore:

```ts
const middle = (lower + (upper - lower) / 2) | 0;
```

Reply:
Gotta love JavaScript gotchas.